gems/sorbet-runtime/test/types/props/constructor.rb

```diff
@@ -53,4 +58,8 @@ class Outer < T::Struct
     o = Outer.new(objs: [Inner.new(i: 4), Inner.new(i: 5)])
     assert_equal([4, 5], o.objs.map(&:i))
   end
+
+  it 'can default untyped fields' do
+    UntypedField.new
+  end
```

Comment:
```suggestion

  it 'can default untyped fields' do
    UntypedField.new
  end
  
  it 'indicates untyped default nil fields are optional'
    assert(T::Props::Utils.optional_prop?(UntypedField.props[:untyped_const]))
    assert(T::Props::Utils.optional_prop?(UntypedField.props[:untyped_prop]))
  end
```

I don't know whether this is the best place to put this test, but we at least need something like this.